service/command.go

```diff
@@ -15,13 +15,57 @@
 package service // import "go.opentelemetry.io/collector/service"
 
 import (
+	"fmt"
+	"os"
+	"strings"
+
 	"github.com/spf13/cobra"
 
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config"
 	"go.opentelemetry.io/collector/confmap"
 	"go.opentelemetry.io/collector/confmap/converter/overwritepropertiesconverter"
 	"go.opentelemetry.io/collector/service/featuregate"
 )
 
+func printComponentSet(set CollectorSettings) {
+	printFactory := func(f component.Factory) {
+		var stability []string
+
+		for _, t := range config.AllDataTypes {
+			if s := f.StabilityLevel(t); s != component.StabilityLevelUndefined {
+				stability = append(stability, fmt.Sprintf("%s: %s", t, s))
+			}
+		}
+
+		if len(stability) == 0 {
+			fmt.Printf("\t%s\n", f.Type())
```

Comment:
Could you use `f.Fprintf` so we can unit test this?

Reply:
Yup.